src/charm.py

```diff
@@ -776,10 +821,32 @@ def _wp_install_cmd(self):
 
     def _wp_install(self):
         """Install WordPress (create WordPress required tables in DB)"""
-        process = self._container().exec(self._wp_install_cmd(), timeout=60)
-        process.wait()
+        logger.debug("Install WordPress, create WordPress related table in the database")
+        msg = ""
+        for _ in range(30):
+            success, msg = self._test_database_connectivity()
+            if success:
+                break
+            time.sleep(self._DB_CHECK_INTERVAL)
+        else:
+            raise exceptions.WordPressInstallError(msg)
+        self.unit.status = ops.model.MaintenanceStatus("Initializing WordPress DB")
+        process = self._container().exec(
+            self._wp_install_cmd(),
+            user=self._WORDPRESS_USER,
+            group=self._WORDPRESS_GROUP,
+            working_dir="/var/www/html",
+            combine_stderr=True,
+            timeout=60,
+        )
+        try:
+            process.wait_output()
+        except ops.pebble.ExecError as e:
+            logger.error(f"WordPress installation failed: %s", e.stdout)
+            raise exceptions.WordPressInstallError("check logs for more information")
```

Comment:
Yes, I think it's best to retry if installation failed with an unexpected reason, maybe defer this event? I can put a `reemit` or `retry` flag in `class WordPressStatusException` to notify the main reconciliation loop to defer the event.

Reply:
But if we just throw that WordPressInstallationException (or other custom exception) so that pebble will get into error status and retry by itself?

Reply:
I think in that case, status message will become "error: x hook failed", which will be a little bit confusing for users.

Reply:
We can set our custom message for the error status too, I think

Reply:
I did some experiments. If an exception was thrown, the status will change to something like `hook failed: "db-relation-changed"` and the traceback and exception message will be printed in debug-log. So that would be less user-friendly because of the vague status message. Just keep this as it is for now and find some other ways to rerun a hook.